### ISSUE TITLE
Piranha.ImageSharp dependency SixLabors.ImageSharp needs updating

### ISSUE THREAD

**Issue by jmbryan4** [PiranhaCMS/piranha.core#2093]
![image](https://github.com/user-attachments/assets/8161e9f9-85cf-4df6-8f42-6586c93cad75)
![image](https://github.com/user-attachments/assets/eb500ab8-54b7-4eac-850f-0c81bfb9a0e0)

**Comment by tidyui** [comment]
There are some major changes to the licensing between v2 and v3 of ImageSharp that we need to clear out before being able to update.

**Comment by usercode** [comment]
Updating the ImageSharp dependency is also important because of breaking changes between version 2 and 3.

I use Piranha CMS with other open source library which target the version 3.x.

If I try to upload an image with the manager, I get this error message.

![grafik](https://github.com/user-attachments/assets/e407a854-f073-42c0-9b8b-2ac2a83af572)

I think there is no license issue. (It's allowed to use ImageSharp as "Transitive Package Dependency" by open source projects like  Piranha CMS.)

https://github.com/SixLabors/ImageSharp/blob/main/LICENSE

**Comment by tidyui** [comment]
@usercode To clarify a bit around the transitive package reference. Yes it’s ok as long as YOUR site isn’t using the ImageSharp package directly and it’s only used from Piranha. If you have any code using ImageSharp and we upgrade to version 3, you would need to buy a license covering your custom usage.

**Comment by usercode** [comment]
> @usercode To clarify a bit around the transitive package reference. Yes it’s ok as long as YOUR site isn’t using the ImageSharp package directly and it’s only used from Piranha. If you have any code using ImageSharp and we upgrade to version 3, you would need to buy a license covering your custom usage.

Yeah, that's true. But what is your conclusion? Keep reference to old ImageSharp version forever? (I don't see the begin of a new major version of Piranha.) Migrate to SkiaSharp?

Maybe you can create a new nuget package called "Piranha.ImageSharp.V3" and everyone can decide which version they want to use.

I fixed it by creating my own Piranha.ImageSharp project like you told. But an official nuget package is nicer. :smile:
